src/agent.ts

```diff
@@ -98,6 +101,14 @@ export default class HttpsProxyAgent extends Agent {
 			socket = net.connect(proxy as net.NetConnectOpts);
 		}
 
+		if (this.timeout) {
+			socket.setTimeout(this.timeout);
+
+			socket.on('timeout', () => {
+				socket.end();
```

Comment:
Do not end the socket on timeout.
The socket will be disconnected automatically through socket.setTimeout and this may cause the application to crash